drivers/adc/adc_cc23x0.c

```diff
@@ -0,0 +1,481 @@
+/*
+ * Copyright (c) 2024 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_cc23x0_adc
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(adc_cc23x0, CONFIG_ADC_LOG_LEVEL);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/util.h>
+
+#include <driverlib/adc.h>
+#include <driverlib/clkctl.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#define ADC_CC23X0_CH_UNDEF	0xff
+#define ADC_CC23X0_CH_COUNT	16
+#define ADC_CC23X0_CH_MAX	(ADC_CC23X0_CH_COUNT - 1)
+
+/* ADC provides four result storage registers */
+#define ADC_CC23X0_MEM_COUNT	4
+#define ADC_CC23X0_MEM_MAX	(ADC_CC23X0_MEM_COUNT - 1)
+
+#define ADC_CC23X0_MAX_CYCLES	1023
+
+#define ADC_CC23X0_INT_MASK	(ADC_INT_MEMRES_00 | \
+				 ADC_INT_MEMRES_01 | \
+				 ADC_INT_MEMRES_02 | \
+				 ADC_INT_MEMRES_03)
+
+#define ADC_CC23X0_INT_MEMRES(i)	(ADC_INT_MEMRES_00 << (i))
+
+#define ADC_CC23X0_MEMCTL(base, i)	HWREG((base) + ADC_O_MEMCTL0 + sizeof(uint32_t) * (i))
+
+static const uint8_t clk_dividers[] = {1, 2, 4, 8, 16, 24, 32, 48};
+
+struct adc_cc23x0_config {
+	const struct pinctrl_dev_config *pincfg;
+	void (*irq_cfg_func)(void);
+	uint32_t base;
+};
+
+struct adc_cc23x0_data {
+	struct adc_context ctx;
+	const struct device *dev;
+	uint32_t res;
+	uint32_t ref_volt[ADC_CC23X0_CH_COUNT];
+	uint32_t channels;
+	uint16_t clk_cycles;
+	uint8_t clk_div;
+	uint8_t ch_sel[ADC_CC23X0_MEM_COUNT];
+	uint8_t mem_index[ADC_CC23X0_CH_COUNT];
+	uint8_t ch_count;
+	uint16_t *buffer;
+	uint16_t *repeat_buffer;
+};
+
+static void adc_context_start_sampling(struct adc_context *ctx)
+{
+	struct adc_cc23x0_data *data = CONTAINER_OF(ctx, struct adc_cc23x0_data, ctx);
+
+	data->channels = ctx->sequence.channels;
+	data->repeat_buffer = data->buffer;
+
+	ADCManualTrigger();
+}
+
+static void adc_context_update_buffer_pointer(struct adc_context *ctx, bool repeat)
+{
+	struct adc_cc23x0_data *data = CONTAINER_OF(ctx, struct adc_cc23x0_data, ctx);
+
+	if (repeat) {
+		data->buffer = data->repeat_buffer;
+	} else {
+		data->buffer += data->ch_count;
+	}
```

Comment:
`data->repeat_buffer` is redundant. Since `data->buffer` is not modified when particular conversion results are stored, you could just have here:
```
	if (!repeat) {
		data->buffer += data->ch_count;
	}
```